typings/styled-components.d.ts

```diff
@@ -0,0 +1,162 @@
+import * as React from "react";
+import { StatelessComponent } from "react";
+
+type ConstrainedProps<C, P> = C & ({ defaultProps?: P } | { new(props?: P, context?: any): any });
+type StyledProps<P> = P & { theme: any };
+
+interface StyledFunction<T, P> {
+  (strs: TemplateStringsArray, ...fns: Array<(props: P) => string>): T;
```

Comment:
Also, if I want to augment props there's no way to do this.
What about adding an overload that accepts additional props type which will be added to styled component?